### ISSUE TITLE
Support token refresh for Composed Task

### ISSUE THREAD

**Issue by nightswimmings** [spring-cloud/spring-cloud-dataflow#4848]
I placed this observation as a closed-US comment (#4600 ) so Id better open a new one.

It looks like composed task does not refresh tokens (not even in Client Credentials Mode, IINW). 
https://github.com/spring-cloud/spring-cloud-dataflow/blob/695507f0cb948a0ddfcb8c62007852d0c8a76454/spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/TaskLauncherTasklet.java#L252
So if the task runs longer thatn your token timeout it dies.

The dataflow-rest-client library (which is a transitive dependency by the very same composed  taks) has added this feature since 2.5.0: https://github.com/spring-cloud/spring-cloud-dataflow/commit/268bb3c87353f005ab3879ea20c873260ff58fd4. Indeed I am really surprised that the OAuth code is duplicated in composed-task instead of using the autoconfigured DataFlowOperations that dataflow-rest-client must be putting in the context (and with latest features, including token refresh for several oauth modes). Is there any reason for this?

**Comment by nightswimmings** [comment]
I tested this in SCDf 2.7.1, taking one of my tasks, setting up an "if(sleep) sleep $TIMEOUT_OF_YOUR_TOKEN+5"
then created a composed-task executing A && A' (A being my debug task) with composed-argument sleep=true and it fails when A completes and it tries to execute A' with "An error occurred while attempting to decode the Jwt: Jwt expired at XX"

**Comment by nightswimmings** [comment]
Ok, it works on 2.8.1. I am really sorry. I did not see any change in the auth code (only move it word by word from one class to another) so I assumed it would work the same in future versions. But as I was desperate I gave it a try and looks like 2.8 fixes. Somewhat the fact that the code is executed in TaskLauncherTasklet instead of DataflowConfiguration fixes it. My guess is that this code is evlauated on every new subtask so that is why it's like refreshing the token as it is requesting it from scratch on each task
